src/liballoc/btree/map.rs

```diff
@@ -783,18 +782,18 @@ impl<K: Ord, V> BTreeMap<K, V> {
     /// map.insert(3, "a");
     /// map.insert(5, "b");
     /// map.insert(8, "c");
-    /// for (&key, &value) in map.range((Included(&4), Included(&8))) {
+    /// for (&key, &value) in map.range((Included(4), Included(8))) {
     ///     println!("{}: {}", key, value);
     /// }
     /// assert_eq!(Some((&5, &"b")), map.range(4..).next());
     /// ```
     #[stable(feature = "btree_range", since = "1.17.0")]
-    pub fn range<T: ?Sized, R>(&self, range: R) -> Range<K, V>
-        where T: Ord, K: Borrow<T>, R: RangeArgument<T>
+    pub fn range<T, R>(&self, range: R) -> Range<K, V>
+        where T: Ord, K: Borrow<T>, R: Into<RangeBounds<T>>
```

Comment:
Shouldn't this be `RangeBounds<&T>`?

Reply:
So I actually recall this being one of the things I struggled with making the original PR. This prevents things like `"a".to_string()..="z".to_string()`. Because `Borrow` works differently from `AsRef`, this prevents both `String` ranges and `str` ranges from working properly.

There may be a way to make it work but I was really stumped in the original PR and got busy and didn't pick it back up.

Reply:
I'm stumped too.